server/src/test/java/com/linecorp/centraldogma/server/internal/storage/repository/git/DiffFormatterTest.java

```diff
@@ -0,0 +1,197 @@
+
+import com.fasterxml.jackson.databind.JsonNode;
+
+import com.linecorp.centraldogma.common.Author;
+import com.linecorp.centraldogma.common.Change;
+import com.linecorp.centraldogma.common.Revision;
+import com.linecorp.centraldogma.server.storage.project.Project;
+
+class DiffFormatterTest {
+
+    private static final String SUMMARY = "summary";
+
+    @TempDir
+    @SuppressWarnings("WeakerAccess")
+    static File repoDir;
+
+    private static GitRepository singleRepo;
+
+    private static GitRepository oldRepo;
+    private static GitRepository newRepo;
+
+    @BeforeAll
+    static void setUp() {
+        singleRepo = createRepo("single");
+        oldRepo = createRepo("old");
+        newRepo = createRepo("new");
+        makeCommits();
+    }
+
+    @AfterAll
+    static void destroy() {
+        singleRepo.internalClose();
+        oldRepo.internalClose();
+        newRepo.internalClose();
+    }
+
+    @ParameterizedTest
+    @ValueSource(strings = {
+            "/**", "/a.json", "/a.txt", "/b/b.json", "/b/b.txt", "/c.json", "/c.txt",
+            "/aa.json", "/non-exist.json",
+    })
+    void testDiff(String pathPattern) throws IOException {
+        //  Single repo:
+        //  Revision     1           2           3              4              5           6              7
+        //  initial commit  Add /a.json              Modify /a.json                           Rename to /aa.json
+        //  initial commit  Add /a.txt
+        //  initial commit              Add /b/b.json
+        //  initial commit              Add /b/b.txt              Remove /b/b.txt
+        //  initial commit                                                         Add /c.json
+        //  initial commit                                                         Add /c.txt
+
+        //  Old repo contains up to the revision 3 commits of Single repo.
+
+        //  Old repo:
+        //  Revision     1           2           3
+        //  initial commit  Add /a.json
+        //  initial commit  Add /a.txt
+        //  initial commit              Add /b/b.json
+        //  initial commit              Add /b/b.txt
+
+        //  New repo is created with the revision 3 of old repo and contains 4 ~ 7 commits of Single repo.
+        //  The revision number - 1 of Single repo is the corresponding revision of New repo.
+
+        //  New repo:
+        //  Revision     1            2                          3              4           5             6
+        //  initial commit  Add /a.json              Modify /a.json                           Rename to /aa.json
+        //  initial commit  Add /a.txt
+        //  initial commit  Add /b/b.json
+        //  initial commit  Add /b/b.txt                           Remove /b/b.txt
+        //  initial commit                                                         Add /c.json
+        //  initial commit                                                         Add /c.txt
```

Comment:
How about using some vertical lines (`|`) to make it clear which change belongs to which commit?